local-tests/test_catch_up.py

```diff
@@ -33,45 +36,45 @@
 
 chain.set_flags_validator('validator')
 
-print('Starting the chain')
+printt('Starting the chain')
 chain.start('aleph')
 
-print('Waiting 90s')
-sleep(90)
-
-finalized_before_kill_per_node = check_finalized(chain)
-print('Killing one validator and one nonvalidator')
+printt('Waiting for finalization')
+chain.wait_for_finalization(0)
+printt('Waiting for authorities')
+chain.wait_for_authorities()
 
-chain[3].stop()
-chain[4].stop()
+printt('Killing one validator and one nonvalidator')
+chain.stop(nodes=[3, 4])
+finalized_before_kill = check_finalized(chain)
 
-print('waiting around 2 sessions')
+printt('Waiting around 2 sessions')
 sleep(30 * 2)
 
-print('restarting nodes')
-finalized_before_start_per_node = check_finalized(chain)
+finalized_before_start = check_finalized(chain)
 
 # Check if the finalization didn't stop after a kill.
-if finalized_before_start_per_node[0] - finalized_before_kill_per_node[0] < 10:
-    print('Finalization stalled')
+if finalized_before_start[0] - finalized_before_kill[0] < 10:
+    printt('Finalization stalled')
     sys.exit(1)
 
+printt('Restarting nodes')
 chain.start('aleph', nodes=[3, 4])
+printt('Waiting for finalization')
+chain.wait_for_finalization(max(finalized_before_start))
 
-print('waiting 45s for catch up')
-sleep(45)
-finalized_after_catch_up_per_node = check_finalized(chain)
+finalized_after = check_finalized(chain)
 
-nonvalidator_diff = finalized_after_catch_up_per_node[4] - finalized_before_start_per_node[4]
-validator_diff = finalized_after_catch_up_per_node[3] - finalized_before_start_per_node[3]
+nonvalidator_diff = finalized_after[4] - finalized_before_start[4]
+validator_diff = finalized_after[3] - finalized_before_start[3]
 
-ALLOWED_DELTA = 5
+delta = 5
 
-# Checks if the murdered nodes started catching up with reasonable nr of blocks.
-if nonvalidator_diff <= ALLOWED_DELTA:
-    print(f"too small catch up for nonvalidators: {nonvalidator_diff}")
+# Check if the murdered nodes started catching up with reasonable nr of blocks.
+if nonvalidator_diff <= delta:
```

Comment:
probably I've some blackout, but why it is _small_ difference that bothers us?

Reply:
Because that means finalization did not restart